0000-settlement-engines/0000-settlement-engines.md

```diff
@@ -0,0 +1,491 @@
+
+If the accounting system or settlement engine receives a request with a **[`Quantity`](#quantity-json-type)** denominated in a unit more precise than its unit, it MUST convert the quantity into its native unit. If so, the resulting amount MUST be rounded down before fulfilling the request.
+
+The response to the request MUST include the converted, rounded **[`Quantity`](#quantity-json-type)** used to fulfill the request, which MUST be less than or equal to the amount sent in the original request. (If the amount rounds down to 0, this this amount MAY be 0.)
+
+Then, the system with the additional precision initiating the request MUST track the leftover sum so it may accumulate and be retried in subsequent requests.
+
+### Settlement Engine HTTP API Endpoints
+
+Settlement engines MUST implement these endpoints.
+
+#### Initiate an account
+
+Inform the settlement engine that a new accounting relationship was instantiated. The settlement engine MAY perform tasks as a prerequisite to settle the new account. For example, a settlement engine implementation might send messages to the peer to exchange ledger identifiers or to negotiate settlement-related fees.
+
+##### Request
+
+```http
+POST /accounts/:id HTTP/1.1
+```
+
+##### Response
+
+```http
+HTTP/1.1 200 OK
+```
+
+#### Delete an account
+
+Instructs the connector that an accounting relationship was terminated.
+
+##### Request
+
+```http
+DELETE /accounts/:id HTTP/1.1
+```
+
+##### Response
+
+```http
+HTTP/1.1 200 OK
+```
+
+#### Perform outgoing settlement
+
+The accounting system requests a settlement to occur with a particular account. The accounting system should [account for the outgoing settlement](#account-for-outgoing-settlements).
+
+The settlement engine serving the request should [asynchronously](#asynchronous-design) perform a [settlement](#settlement).
+
+##### Request
+
+```http
+POST /accounts/:id/settlements HTTP/1.1
+Accept: application/json
+Content-Type: application/json
+Idempotency-Key: <key>
+```
+
+> **[`Quantity`](#quantity-json-type)** to settle
+
+##### Response
+
+```http
+HTTP/1.1 202 ACCEPTED
+Content-Type: application/json
+```
+
+> **[`Quantity`](#quantity-json-type)** enqueued to settle, which MUST be less than or equal to the quantity in the original request
+
+#### Handle incoming message
+
+Incoming messages from the peer should prompt requests to this endpoint so the settlement engine may handle them. The settlement engine can respond with its own message, which should be proxied back to the peer it originated from.
+
+##### Request
+
+```http
+POST /accounts/:id/messages HTTP/1.1
+Accept: application/octet-stream
+Content-Type: application/octet-stream
+```
+
+> _&lt;raw bytes of message&gt;_
+
+##### Response
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/octet-stream
+```
+
+> _&lt;raw bytes of response&gt;_
+
+#### Create or update incoming settlement webhook
+
+##### Request
+
+```http
+POST /accounts/:id/webhooks/settlements
```

Comment:
Couldn't this just be part of the account record you create with `PUT /accounts/:id`? You would use the same command to create the account as to change it

This also seems better so that you don't need 3 HTTP requests to set up an account

_And_, all accounts MUST have these callback endpoints configured so it makes more sense to make it a required part of the account details that you can update but not remove